server/server/src/sync/translation_central/report.rs

```diff
@@ -0,0 +1,109 @@
+use crate::sync::{
+    sync_serde::empty_str_as_option, translation_central::TRANSLATION_RECORD_REPORT,
+};
+use repository::{CentralSyncBufferRow, ReportContext, ReportRow, ReportType};
+
+use serde::{Deserialize, Serialize};
+
+use super::{CentralPushTranslation, IntegrationUpsertRecord};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq)]
+pub enum LegacyReportEditor {
+    #[serde(rename = "omsupply")]
+    OmSupply,
+    #[serde(other)]
+    Others,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq)]
+pub enum LegacyReportContext {
+    #[serde(rename = "Customer Invoice")]
+    CustomerInvoice,
+    #[serde(rename = "Supplier Invoice")]
+    SupplierInvoice,
+    #[serde(rename = "Requisition")]
+    Requisition,
+    #[serde(rename = "Stock Take")]
+    Stocktake,
+
+    #[serde(other)]
+    Others,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize)]
+pub struct LegacyReportRow {
+    #[serde(rename = "ID")]
+    pub id: String,
+    pub report_name: String,
+    pub editor: LegacyReportEditor,
+    pub context: LegacyReportContext,
+    pub template: String,
+
+    #[serde(deserialize_with = "empty_str_as_option")]
+    #[serde(rename = "Comment")]
+    pub comment: Option<String>,
+}
+
+pub struct ReportTranslation {}
+impl CentralPushTranslation for ReportTranslation {
+    fn try_translate(
+        &self,
+        sync_record: &CentralSyncBufferRow,
+    ) -> Result<Option<IntegrationUpsertRecord>, anyhow::Error> {
+        let table_name = TRANSLATION_RECORD_REPORT;
+        if sync_record.table_name != table_name {
+            return Ok(None);
+        }
+
+        let data = serde_json::from_str::<LegacyReportRow>(&sync_record.data)?;
+
+        let t = match data.editor {
```

Comment:
a bit of a :nit, I think the scope of this variable is slightly too big for single variable name, r#type, 

Also, do we need report type in omsupply database ?

Reply:
thanks [81be293](https://github.com/openmsupply/open-msupply/pull/147/commits/81be2932e69a332d51e710af72e0d1cd4388454a)

There is only one type right now used but the template data depends on the type for which reason I think its important to already have it in. I added a doc comment.